asu/build.py

```diff
@@ -57,14 +58,25 @@ def build(build_request: BuildRequest, job=None):
         f"Container version: {container_version_tag} (requested {build_request.version})"
     )
 
+    mounts: list[dict[str, Union[str, bool]]] = []
+    environment: dict[str, str] = {}
+
     image = f"{settings.base_container}:{build_request.target.replace('/', '-')}-{container_version_tag}"
 
+    if settings.squid_cache and build_request.version.lower().endswith("snapshot"):
+        image = "localhost/imagebuilder:setup"
+        environment.update(
+            {
+                "TARGET": build_request.target,
+                "use_proxy": "on",
```

Comment:
If we also want to include release snapshots:
```
    "VERSION_PATH": "snapshots" if build_request.version == "SNAPSHOT" else f"releases/{build_request.version}",
```

Reply:
We could but then again it kind of adds extra unpacking costs each time. Not sure if worth it.

Reply:
This one addresses the
```
    ... build_request.version.lower().endswith("snapshot"):
```
a few lines above, which is going to try to do the release snapshots.  Should that be `version == "snapshot"` to just do main snapshots and ignore release?

Reply:
Right, but its `VERSION_PATH` is under `releases/...`, not `snapshots`, which is the assumed default in `setup.sh` line 7, so you'll get download errors on setup.sh's wgets

Reply:
Right! Mind creating a PR?